src/hotspot/share/runtime/javaThread.cpp

```diff
@@ -1179,7 +1179,7 @@ void JavaThread::set_is_disabler_at_start(bool val) {
 //
 bool JavaThread::java_suspend(bool register_vthread_SR) {
   // Suspending a vthread transition disabler can cause deadlocks.
-  assert(!is_vthread_transition_disabler(), "no suspend allowed for vthread transition disablers");
```

Comment:
So except for self-suspend, how can we hit this assert if we use an exclusive disabler? And if we remove this assert and allow a disabler to self-suspend, wouldn't we deadlock since the resumer would be unable to run?
I can see why we need the change in `HandshakeState::get_op_for_self` since a thread could be suspended outside a disable operation (e.g. waiting in `disable_transition_for_one()`) and then try to process it inside the disable operation, but I don't get this assert removal.